packages/fetchai/skills/registration_aw1/behaviours.py

```diff
@@ -43,24 +43,28 @@ def setup(self) -> None:
         :return: None
         """
         strategy = cast(Strategy, self.context.strategy)
-        signing_dialogues = cast(SigningDialogues, self.context.signing_dialogues)
-        msg, _ = signing_dialogues.create(
-            counterparty=self.context.decision_maker_address,
-            performative=SigningMessage.Performative.SIGN_MESSAGE,
-            raw_message=RawMessage(
-                strategy.ledger_id, strategy.ethereum_address.encode("utf-8")
-            ),
-            terms=Terms(
-                ledger_id=strategy.ledger_id,
-                sender_address="",
-                counterparty_address="",
-                amount_by_currency_id={},
-                quantities_by_good_id={},
-                nonce="",
-            ),
-        )
-        self.context.logger.info("sending signing_msg to decision maker...")
-        self.context.decision_maker_message_queue.put_nowait(msg)
+        if strategy.announce_termination_key is not None:
+            self.context.shared_state[strategy.announce_termination_key] = False
```

Comment:
What do we need this for?

Reply:
if `strategy.announce_termination_key` is not `None`, it means we want this skill to announce when it finishes its job (so another skill which needs to start strictly after this one knows when to start). 

The `announce_termination_key` is the key in the `shared_state` that both these skills try to write to/read from.

Here (one of the first codes executed by this skill) we are adding this key in the `shared_state` with `False` to announce this skill hasn't finished yet so when the other skill reads the value of this key from `shared_state`
it is there with value False. 
This value changes to True when the skill finishes (handler `_handle_success`).